### DISCUSSION TITLE
FileUpload with mp3 doesn't preview.

### DISCUSSION THREAD

**Discussion by TheFrankman** [filamentphp/filament#10049] [Help]
When uploading an audio asset everything is fine. The audio file can be played and uploaded.

![image](https://github.com/filamentphp/filament/assets/1402017/b1a8628c-583c-4af3-92fe-d364912282f0)

However, when you visit the audio asset in edit mode, or in view mode the audio asset does not have a play button : 

![image](https://github.com/filamentphp/filament/assets/1402017/adeda2c6-68d2-4eec-8e6d-d5b86cf195f2)

Is this a known issue or am i doing something wrong. I have the exact same functionality for my artwork which works fine, but just audio assets don't preview. For some more context here is my form input : 

```
FileUpload::make('artwork')
            ->label('Upload new Audio')
            ->storeFileNamesIn('original_name')
            ->visibility('public')
            ->disk('public')
            ->acceptedFileTypes(['audio/mpeg'])
            ->directory("{$organisation->slug}/audio_assets")
            ->getUploadedFileNameForStorageUsing(
                function(TemporaryUploadedFile $file): string {
                    $fileName = $file->getClientOriginalName();
                    $extension = pathinfo($fileName, PATHINFO_EXTENSION);
                    $name = pathinfo($fileName, PATHINFO_FILENAME);
                    $slug = Str::slug($name, '-');

                    $timestamp = now()->timestamp;
                    return "{$timestamp}-{$slug}.{$extension}";
                }
            )
            ->statePath('path');
```

And here is an image of my schema : 

![image](https://github.com/filamentphp/filament/assets/1402017/3ae7bdae-4aca-49e1-b83d-75b64525438a)

**Comment by leandrocfe** [comment]
There isn't a play button I think, but you can use 
```php
->openable()
->downloadable()
```

**Reply by TheFrankman** [reply]
openable() works which is good because it proves there isn't anything wrong with the way i'm storing the asset, and at least gives the users somewhere to go. But very odd that it's capable of playing during upload but not in edit/view. I feel this must be a bug right ?

**Reply by leandrocfe** [reply]
not sure, because I don't know if the audio file has `previewable()`. You can open an issue with a reproduction repo if you want

**Reply by TheFrankman** [reply]
Is there a guide for providing a reproduction repo that you know of ?
It's obviously quite an undertaking as you need to supply an env and boot instructions etc.
